services-turf/src/test/java/com/mapbox/turf/TurfConversionTest.java

```diff
@@ -78,6 +80,129 @@ public void convertDistance() throws TurfException {
+              Point.fromLngLat(61.938950426660604, 5.9765625))))),
+          Feature.fromGeometry(Polygon.fromLngLats(Arrays.asList(
+            Arrays.asList(
+              Point.fromLngLat(11.42578125, 16.636191878397664),
+              Point.fromLngLat(7.91015625, -9.102096738726443),
+              Point.fromLngLat(31.113281249999996, 17.644022027872726),
+              Point.fromLngLat(11.42578125, 16.636191878397664)
+            ))))
+        });
+
+    FeatureCollection featureCollectionWithNewMultiPolygonObject = TurfConversion.combine(polygonFeatureCollection);
+    assertNotNull(featureCollectionWithNewMultiPolygonObject);
+
+    MultiPolygon multiPolygon = (MultiPolygon) featureCollectionWithNewMultiPolygonObject.features().get(0).geometry();
+    assertNotNull(multiPolygon);
+
+    // Checking the first Polygon in the MultiPolygon
+
+    // Checking the first Point
+    assertEquals(61.938950426660604, multiPolygon.coordinates().get(0).get(0).get(0).longitude(), DELTA);
+    assertEquals(5.9765625, multiPolygon.coordinates().get(0).get(0).get(0).latitude(), DELTA);
+
+    // Checking the second Point
+    assertEquals(52.696361078274485, multiPolygon.coordinates().get(0).get(0).get(1).longitude(), DELTA);
+    assertEquals(33.046875, multiPolygon.coordinates().get(0).get(0).get(1).latitude(), DELTA);
+
+    // Checking the second Polygon in the MultiPolygon
+
+    // Checking the first Point
+    assertEquals(11.42578125, multiPolygon.coordinates().get(1).get(0).get(0).longitude(), DELTA);
+    assertEquals(16.636191878397664, multiPolygon.coordinates().get(1).get(0).get(0).latitude(), DELTA);
+
+    // Checking the second Point
+    assertEquals(7.91015625, multiPolygon.coordinates().get(1).get(0).get(1).longitude(), DELTA);
+    assertEquals(-9.102096738726443, multiPolygon.coordinates().get(1).get(0).get(1).latitude(), DELTA);
+  }
+
+  // TODO: Add test that checks Feature amount
+  @Test
+  public void geometryTypeMixThrowsException() throws TurfException {
+    thrown.expect(TurfException.class);
+    thrown.expectMessage(startsWith("Your FeatureCollection must be of all of the same geometry type."));
+
+    // Create a FeatureCollection with a Point Feature and a Polygon Feature
+    FeatureCollection pointAndPolygonFeatureCollection =
+      FeatureCollection.fromFeatures(
+        new Feature[]{
+          Feature.fromGeometry(Point.fromLngLat(-2.46,
+            27.6835)),
+          Feature.fromGeometry(Polygon.fromLngLats(Arrays.asList(
+            Arrays.asList(
+              Point.fromLngLat(11.42578125, 16.636191878397664),
+              Point.fromLngLat(7.91015625, -9.102096738726443),
+              Point.fromLngLat(31.113281249999996, 17.644022027872726),
+              Point.fromLngLat(11.42578125, 16.636191878397664)
+            ))))
+        });
+
+    // Building a geometry with this FeatureCollection should through an error
+    TurfConversion.combine(pointAndPolygonFeatureCollection);
+  }
+
```

Comment:
tests look good but we also want to port the js tests as closely as possible:
https://github.com/Turfjs/turf/blob/master/packages/turf-combine/test.js
Porting these is helpful as they act as integration tests that validate our implementation.

Reply:
Latest commit adds and refactors tests to match https://github.com/Turfjs/turf/blob/master/packages/turf-combine/test.js#L154